File: 01_GOF23/DP_examples/Lua/03_Behavioral/Memento/Memento.lua
-- 备忘录类
local Memento = {}
Memento.__index = Memento

function Memento:new(version)
    local o = {}
    setmetatable(o, self)
    o.states = {}
    o.Version = version
    return o
end

function Memento:GetState(identifier)
    return self.states[identifier]
end

function Memento:SetState(identifier, state)
    self.states[identifier] = state
end

return Memento

File: 01_GOF23/DP_examples/Python/03_Behavioral/Memento/memento.py
# 备忘录类
class Memento:
    def __init__(self, version):
        self.states = {}
        self.version = version
    
    def get_state(self, identifier):
        return self.states.get(identifier)
    
    def set_state(self, identifier, state):
        self.states[identifier] = state
